### PR TITLE
Use production ready service mesh proxy image for Kourier

### PR DESCRIPTION
**Pull request by pierDipi**
Use image on registry.redhat.io [1] vs quay.io/maistra image

[1] https://catalog.redhat.com/software/containers/openshift-service-mesh/proxyv2-rhel8

### CODE REVIEW

File: olm-catalog/serverless-operator/project.yaml [hunk 1]
```diff
@@ -1,75 +1,75 @@
...
+  serving: knative-v1.15
+  # serving midstream branch name
+  serving_artifacts_branch: release-v1.15
+  # versions for networking components
+  kourier: knative-v1.15
+  net_kourier_artifacts_branch: release-v1.15
+  net_istio: knative-v1.15
+  net_istio_artifacts_branch: release-v1.15
+  redhat-knative-istio-authz-chart: 1.35.0
+  service_mesh_proxy: registry.redhat.io/openshift-service-mesh/proxyv2-rhel8@sha256:02b834fd74da71ec37f6a5c0d10aac9a679d1a0f4e510c4f77723ef2367e858a
```
Comment (ReToCode) [inline olm-catalog/serverless-operator/project.yaml:44]: I assume that hash is equivalent with 2.4-latest?

File: olm-catalog/serverless-operator/project.yaml [hunk 2]
```diff
@@ -1,75 +1,75 @@
 project:
-    name: serverless-operator
-    # When bumping the Operator to a new version (major and minor), make sure to also update
-    # all components in `dependencies.previous` to the same versions as `dependencies` in the same PR.
-    # Otherwise, the upgrade tests will not pass, as we have a different SO version with the same bundle contents.
-    # Also make sure to update values under `olm.previous` by copying from `olm.replaces` and `olm.skipRange`.
-    version: 1.35.0
+  name: serverless-operator
```
Comment (ReToCode) [inline olm-catalog/serverless-operator/project.yaml:2]: The whitespace changes were probably not intended? Or where they?

**Review by ReToCode — APPROVED** [review]
LGTM other than the nits/Qs.

File: olm-catalog/serverless-operator/project.yaml (hunk 1 above)
Comment (pierDipi) [reply olm-catalog/serverless-operator/project.yaml:44]: See discussion here: https://redhat-internal.slack.com/archives/CKR568L8G/p1727254666855229

**Review by ReToCode — APPROVED** [review]
/lgtm
/approve

**Comment by skonto** [comment]
We used to have the developer version of it so we catch issues early. [Here](https://redhat-internal.slack.com/archives/C019EPZ233P/p1728027911063869) is a question I asked recently to SM folks.

**Comment by ReToCode** [comment]
IMHO we build the production images from this repo now, so we also need to have the production images referenced :) Seems like chicken-egg?

**Pull request merged**